doc/stack_vs_cabal_config.md

```diff
@@ -0,0 +1,135 @@
+<div class="hidden-warning"><a href="https://docs.haskellstack.org/"><img src="https://rawgit.com/commercialhaskell/stack/master/doc/img/hidden-warning.svg"></a></div>
+
+# stack vs cabal config files
```

Comment:
The title is confusing. This article doesn't describe `~/.stack/config.yaml` and `~/.cabal/config` differences.

*stack.yaml and pkgname.cabal files*.

To justify *versus*, you could write *stack.yaml vs cabal.project* article, but that's completely different.

Reply:
I'm having a really difficult time understanding what you're trying to say here. To be honest, I'm failing to understand just about any of the comments you've put on this PR.

Reply:
The title of the page is confusing.

- *cabal config* is easily misinterpret as `cabal.config` files. IMHO *cabal package file* is better.
- *stack config* may mean `~/.stack/config.yaml`. IMO *stack.yaml* is better.

TL;DR

```diff
-# stack vs cabal config files
+# stack.yaml vs cabal package files
```